### PR TITLE
Fix console.* calls on Windows

### LINKED ISSUE #3032: console.log outputs in main process are not printed on Windows
Quoted from https://github.com/atom/electron/pull/2974#issuecomment-146342860:

> Keep in mind that on Windows, calls to console.* won't be printed, but they will be printed on Linux and OS X. The reason is that for Windows, console.* is implemented on top of the logging mechanism (see details here).

### PR DESCRIPTION
**Pull request by etiktin**
- Calls for `console.*` on browser process are printed with no need for `--enable-logging`
- The output is without the logging prefix
- The caret in the terminal is always after the last output
- The first output starts on a new line and not at the prompt
- console.* from renderer are not printed to cmd
- Added a missing '\n' in the default_app help output

This resolves #3032, except for making sure that error logs are not printed when running without `--enable-logging`.

Before:
before
After:
after

### CODE REVIEW

File: atom/browser/lib/init.coffee [hunk 1]
```diff
@@ -22,7 +22,7 @@ if process.platform is 'win32'
   # Redirect node's console to use our own implementations, since node can not
   # handle console output when running as GUI program.
   print = (args...) ->
-    process.log util.format(args...)
+    process.log util.format(args...) + "\n"
   console.log = console.error = console.warn = print
   process.stdout.write = process.stderr.write = print
```
Comment (zcbenz) [inline atom/browser/lib/init.coffee]: The `process.stdout.write` should behave the same with `std::wcout`, e.g. not formatting the arguments and append no new line. And we can also make it accept `Buffer` as input as mentioned in #2390.

Comment (etiktin) [inline atom/browser/lib/init.coffee]: Fixed according to your suggestions.

**Comment by zcbenz** [comment]
Awesome, thanks!

**Comment by etiktin** [comment]
@zcbenz, I just checked what would happen if we would remove the special treatment of console.* and process.std*.write and let node handle it and everything worked. It even supported unicode correctly, but no colors.
Could it be that some update to node fixed the old issues that made us replace it? Do you have some code I can try to reconstruct the old issues?

**Comment by zcbenz** [comment]
It seems to depend on how you run Electron, like from Console, or from a cygwin shell, or by clicking the program in Explorer.

I remember starting Electron by double-clicking will have the issue, because it doesn't have a console attached and Node will throw exception. But since we are suppressing that exception in recent versions, it is probably fine to remove the hacks.

**Pull request merged**